app/assets/javascripts/transactions.js

```diff
@@ -51,26 +46,41 @@ $(function () {
     $('#new_transaction').show();
   });
 
+  $(document).on('change', '#transfer_amount', function(e) {
+    prep_rate_and_hints(exchange_rate = false, hints = true)
+  });
+
+  $(document).on('change', '#transfer_exchange_rate', function(e) {
+    prep_rate_and_hints(exchange_rate = false, hints = true)
+  });
+
   $(document).on('change', '#transfer_bank_account_id', function(e) {
-    if ($('#transfer_exchange_rate').size()) {
-      show_hide_exchange_rate();
-    }
+    prep_rate_and_hints(exchange_rate = true, hints = true)
   });
 
   $(document).on('change', '#transfer_reference_id', function(e) {
-    if ($('#transfer_exchange_rate').size()) {
-      show_hide_exchange_rate();
-    }
+    prep_rate_and_hints(exchange_rate = true, hints = true)
   });
 });
 
-function show_hide_exchange_rate() {
-  fromCurr = $('#transfer_bank_account_id option:selected').attr('data_currency');
-  toCurr = $('#transfer_reference_id option:selected').attr('data_currency')
+function prep_rate_and_hints(exchange_rate, hints) {
+  var fromCurr = $('#transfer_bank_account_id option:selected').parent().attr('label');
```

Comment:
@AlexanderUlitin variables will be undefined which is ok(I think) since it will be checked in function later